### PR TITLE
Fixing OPT fast tokenizer option.

### PR DESCRIPTION
**Pull request by Narsil**
# What does this PR do?

Fixes the relevant issues: 

https://huggingface.co/wjmcat/opt-350m-paddle/discussions/1
https://huggingface.slack.com/archives/C01N44FJDHT/p1653511495183519 (internal link)
https://github.com/huggingface/transformers/pull/17088#discussion_r871246439

Basically, the OPT tokenizer is a GPT2 tokenizer that adds a BOS token at the start
of the tokens.
Lots of back&forth at the time, but the truth is that the `ByteLevel(trim_offsets=False)` 
post_processor, actually doesn't do anything, so we can just replace it with a simple
`TemplateProcessing` processor and everything works correctly.

This PR fixes the biggest culprit (missing BOS token on the fast tokenizer version).

Call to witness on other issues I might have missed.

@SauLu
@patrickvonplaten
@mishig

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by SaulLu** [comment]
Good point! 

Indeed, I had not noticed at all that `ByteLevel(trim_offsets=False)` was not doing anything. As `trim_offset` is not an argument in the `__init__` either, I don't see any problem that your solution could cause! :blush:

File: tests/models/gpt2/test_tokenization_gpt2.py [hunk 1]
```diff
@@ -250,3 +250,27 @@ def test_add_bos_token_slow(self):
     # tokenizer has no padding token
     def test_padding_different_model_input_name(self):
         pass
+
+
+@require_tokenizers
+class OPTTokenizationTest(unittest.TestCase):
```
Comment (SaulLu) [inline tests/models/gpt2/test_tokenization_gpt2.py:256]: Out of curiosity, why create a new test class?  :hugs:

Comment (Narsil) [reply tests/models/gpt2/test_tokenization_gpt2.py:256]: No mixin, it's a simple test.

If there are better/more suited locations for this test I can move it.
I just thought it didn't fit the Mixin type of tests (which are intended to by highly generic, right ?)

Comment (patrickvonplaten) [reply tests/models/gpt2/test_tokenization_gpt2.py:256]: Fine for me - think in the normal GPT2 Fast Test we test the tokenizer without leading BOS token. @Narsil could we maybe add 1,2 more tests here to check that fast gives identical output to slow and maybe also a test that the first token (BOS TOKEN) can be changed to whatever token the user wants with save / re-load?

**Review by patrickvonplaten — APPROVED** [review]
Wuhuu super cool PR @Narsil thanks a mille! 

If not too much of a hussle it'd be very nice to add 2,3 more tests for the fast OPT tokenizer

File: tests/models/gpt2/test_tokenization_gpt2.py (hunk 1 above)
Comment (Narsil) [reply tests/models/gpt2/test_tokenization_gpt2.py:256]: I added two tests, are those what you had in mind ?

**Review by patrickvonplaten — APPROVED** [review]
Thanks!

**Pull request merged**